spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -647,7 +647,7 @@ private static void deleteManifests(FileIO io, List<ManifestFile> manifests) {
     private final String format;
 
     public SparkPartition(Map<String, String> values, String uri, String format) {
-      this.values = ImmutableMap.copyOf(values);
+      this.values = new HashMap<>(values);
```

Comment:
Can you use `Maps.newHashMap`? We prefer the Guava factory methods to individual implementations.

Also, can you add a test for this that does a round-trip serialization using Kryo?